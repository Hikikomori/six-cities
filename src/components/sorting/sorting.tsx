import {connect} from "react-redux";

import {ActionCreator as baseAction, Operation as baseOperation} from "../../reducer/base/base";

const {toggleSortingFlag} = baseAction;

const {setSorting} = baseOperation;

interface Props {
  options: Array<string>,
  active: string,
  setSorting: (sorting: string) => void,
  isSortingOpened: boolean,
  toggleSortingFlag: () => void
}

const Sorting: React.FC<Props> = (props) => {
  const {
    options,
    active,
    setSorting,
    isSortingOpened,
    toggleSortingFlag} = props;

  return <div className="places__sorting">
    <span className="places__sorting-caption">Sort by</span>
    <span className="places__sorting-type" tabIndex={0} onClick={toggleSortingFlag}>
                  &nbsp;{active}
      <svg className="places__sorting-arrow" width="7" height="4">
                <use xlinkHref="#icon-arrow-select" />
              </svg>
            </span>
    <ul className={`${isSortingOpened ? `places__options--opened` : ``} places__options places__options--custom`}>
      {options.map((option, i) => {
        return <li className={`${active === option ? `places__option--active` : ``} places__option`} onClick={() => setSorting(option)} tabIndex={0} key={i}>{option}</li>
      })}
    </ul>
  </div>
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  options: state.BASE.sortingOptions,
  active: state.BASE.sorting,
  isSortingOpened: state.BASE.isSortingOpened,
});

const mapDispatchToProps = {
  setSorting,
  toggleSortingFlag,
};

export {Sorting};

export default connect(mapStateToProps, mapDispatchToProps)(Sorting)
