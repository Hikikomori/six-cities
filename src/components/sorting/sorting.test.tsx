import * as renderer from 'react-test-renderer';
import {Sorting} from './sorting';

it(`Sorting correctly renders`, () => {
  const tree = renderer
    .create(<Sorting
      options={['Popular']}
      active={'Popular'}
      isSortingOpened={true}
      toggleSortingFlag={jest.fn()}
      setSorting={jest.fn()}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
