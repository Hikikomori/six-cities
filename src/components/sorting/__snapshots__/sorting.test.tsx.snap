// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorting correctly renders 1`] = `
<div
  className="places__sorting"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[MockFunction]}
    tabIndex={0}
  >
    Â 
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options--opened places__options places__options--custom"
  >
    <li
      className="places__option--active places__option"
      onClick={[Function]}
      tabIndex={0}
    >
      Popular
    </li>
  </ul>
</div>
`;
